### ISSUE TITLE
apps inherit app_name even when new name is specified

### ISSUE THREAD

**Issue by shoup-healthtap** [thuehlinger/daemons#55]
I'm trying to run multiple groups of daemons, each group with a different name. The daemons run correctly, but they inherit their name from the first group to run even when I explicitly specify a new name.

example code:
    options1 = {
      multiple: true,
      app_name: task1
    }
    options2 = {
      multiple: true,
      app_name: task2
    }
    2.times do Daemons.call(options1) { doTask1() } end
    2.times do Daemons.call(options2) { doTask2() } end

The above code correctly spawns the four tasks but incorrectly refers to all four by the name `task1`.

Looking at the source code, I believe the issue is [here](https://github.com/thuehlinger/daemons/blob/master/lib/daemons/application.rb#L43) in `Application.new` where new applications inherit the name of their group.

Is there a better way to use Daemons such these tasks are created within different groups? (Is this intended behavior?) Or is this a bug?

**Comment by thuehlinger** [comment]
Yes, this is the intended behaviour, since the simple `#call` interface assumes that you only want one application group.
I suggest to create different `ApplicationGroup`s yourself and use these to create your applications. Like so:
`
options[:proc] = myproc1
group1 = ApplicationGroup.new(app_name, options)
group1.new_application(options)

options[:proc] = myproc
group2 = ApplicationGroup.new(app_name, options)
group2.new_application(options)`
